odoo/orm/environments.py

```diff
@@ -999,7 +1014,7 @@ def process(model, field, field_cache):
             model = env[field.model_name]
             if field in depends_context:
                 for context_keys, inner_cache in field_cache.items():
-                    context = dict(zip(depends_context[field], context_keys))
+                    context: dict[str, typing.Any] = dict(zip(depends_context[field], context_keys))  # type: ignore
```

Comment:
If the linter ignores the type, then it's useless.
```suggestion
                    context = dict(zip(depends_context[field], context_keys))  # type: ignore
```

Reply:
I fixed the type in registry (collectors) and adapted the code.